ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1331,6 +1331,21 @@ public Table getTable(final String dbName, final String tableName) throws HiveEx
     }
   }
 
+  /**
+   * Returns metadata of the table
+   *
+   * @param tableName
+   *          the tableName object
+   * @return the table
+   * @exception HiveException
+   *              if there's an internal error or if the table doesn't exist
+   */
+  public Table getTable(TableName tableName) throws HiveException {
+
+    return tableName.getDb() == null ? this.getTable(tableName.getTable(), true) : this
```

Comment:
a table without db is not invalid? is it only needed temporarilt - until this whole thing is done?